File: nitpickers/pyql/pyql/ast/form/if_statement.py
from pyql.ast.form import statement
from pyql.ast.form import block
from pyql.ast.expression import expressions


class IfStatement(statement.Statement):

    def __init__(self, location, expression: expressions.Expression, block: block.Block):
        super().__init__(location)
        self._expression = expression
        self._block = block

    @property
    def expression(self):
        return self._expression

    @property
    def block(self):
        return self._block

    def __repr__(self):
        return "AST IfStatement at: " + str(self.location)

File: Abel_Elias/src/classes/statements/IfStatement.java
package classes.statements;

import classes.Block;
import classes.CodeBlock;
import classes.expressions.BooleanExpression;
import classes.expressions.Expression;

import javax.naming.ldap.ExtendedRequest;

public class IfStatement extends Statement {

    private BooleanExpression expression;
    private Block block;
    private CodeBlock code;

    public IfStatement(BooleanExpression expression, CodeBlock code, Block block) {
        super(code);
        this.expression = expression;
        this.code = code;
        this.block = block;
    }

    public BooleanExpression getExpression() {
        return expression;
    }

    public CodeBlock getCode() {
        return code;
    }
    public Block getBlock() {return block;}

    @Override
    public void accept(StatementVisitor visitor) {
        visitor.visitIfStatementWithExpr(this, expression);
    }

    @Override
    public void accept(StatementVisitor visitor, BooleanExpression expression) {
        visitor.visitIfStatementWithExpr(this, expression);
    }
}
